hathor/event/storage/memory_storage.py

```diff
@@ -33,3 +33,10 @@ def get_event(self, key: int) -> Optional[BaseEvent]:
             return None
 
         return self.events[key]
+
+    def get_last_event(self) -> Optional[BaseEvent]:
+        if not self.events:
+            return None
+
+        last_id = list(self.events)[-1]
```

Comment:
That's O(n). I feel we should store the last event in a private attribute (e.g., `_last_event`) to make this O(1).

Reply:
I'll fix this. I'm also pulling `_last_group_id` from a later PR, it uses the same strategy that you suggested.

Reply:
Fixed.

Reply:
Great! I'll review it again. Thanks!